File: src/main/kotlin/nl/appetit/api/logic/model/Product.kt
package nl.appetit.api.logic.model

import java.math.BigDecimal
import java.time.Instant

data class Product(
    val id: Int? = null,
    val name: String,
    val price: BigDecimal,
    val description: String? = null,
    val imageUrl: String = "",
    val isAvailable: Boolean,
    val categoryId: Int? = null,
    val createdAt: Instant = Instant.now(),
    val updatedAt: Instant = Instant.now()
)

File: frontend/lib/interfaces/product.ts
import { Tag } from './tag';

export interface Product {
  id: number;
  name: string;
  price: number;
  description: string;
  imageUrl: string;
  available: boolean;
  categoryId: number | null;
  tags?: Tag[];
  createdAt: string;
  updatedAt: string;
}